sos/report/plugins/ceph_mgr.py

```diff
@@ -103,17 +72,74 @@ def setup(self):
             "version"
         ]
 
-        self.add_cmd_output([
-            f"ceph daemon {m} {cmd}" for m in self.get_socks() for cmd in cmds]
-        )
-
-    def get_socks(self):
+        if not microceph_pkg:
+            self.add_file_tags({
+                '/var/log/ceph/(.*/)?ceph-mgr.*.log': 'ceph_mgr_log',
+            })
+
+            self.add_forbidden_path([
+                "/etc/ceph/*keyring*",
+                "/var/lib/ceph/**/*keyring*",
+                "/var/lib/ceph/**/osd*",
+                "/var/lib/ceph/**/mon*",
+                # Excludes temporary ceph-osd mount location like
+                # /var/lib/ceph/tmp/mnt.XXXX from sos collection.
+                "/var/lib/ceph/**/tmp/*mnt*",
+                "/etc/ceph/*bindpass*",
+            ])
+
+            self.add_copy_spec([
+                "/var/log/ceph/**/ceph-mgr*.log",
+                "/var/lib/ceph/**/mgr*",
+                "/var/lib/ceph/**/bootstrap-mgr/",
+                "/run/ceph/**/ceph-mgr*",
+            ])
+
+            self.add_cmd_output(
+                [f"ceph {cmd}" for cmd in ceph_mgr_cmds])
+            # get ceph_cmds again as json for easier automation parsing
+            self.add_cmd_output(
+                [f"ceph {cmd} --format json-pretty" for cmd in ceph_mgr_cmds],
+                subdir="json_output",
+            )
+
+            self.add_cmd_output([
+                f"ceph daemon {m} {cmd}" for m in self.get_socks(
+                    '/var/run/ceph') for cmd in cmds]
+            )
+        else:
+            self.add_file_tags({
+                '/var/log/ceph/common/logs/ceph-mgr.*.log': 'ceph_mgr_log',
+            })
+
+            self.add_forbidden_path([
+                "/var/snap/microceph/common/**/*keyring*",
+            ])
+
+            self.add_copy_spec([
+                "/var/snap/microceph/common/logs/ceph-mgr*.log",
+            ])
+
+            self.add_cmd_output(
+                [f"ceph {cmd}" for cmd in ceph_mgr_cmds])
+            # get ceph_cmds again as json for easier automation parsing
+            self.add_cmd_output(
+                [f"ceph {cmd} --format json-pretty" for cmd in ceph_mgr_cmds],
+                subdir="json_output",
+            )
+
+            self.add_cmd_output([
+                f"ceph daemon {m} {cmd}" for m in self.get_socks(
+                    '/var/snap/microceph') for cmd in cmds]
+            )
```

Comment:
minor nit, what about using a variable for the directory, and then have the `add_cmd_output(` aspect of it in the following way?

```python
            self.add_cmd_output([
                f"ceph daemon {m} {cmd}" for m in self.get_socks(
                    directory) for cmd in cmds]
            )
```

Maybe also do the common bits outside of the if statement, wdyt?

Reply:
done, thanks, thats a good cleanup